src/breedsRepo.js

```diff
@@ -1,82 +1,170 @@
+        return getProm();
     }
 
-    findIndexById(id) {
-        const breedsCopy = this.getAllBreeds; 
-        const foundBreed = breedsCopy.findIndex((breed) => {
-            return breed.breedId === id;       
-        });
-        return foundBreed;  // foundBreed is either -1 or a valid index
+    // member function used in updateBreedById and deleteBreedById - returns true or false 
+    checkForId(id) {
+
+        const dao2 = this.dao;
+
+        const getProm = async function () {
+            try {
+                const isId = await dao2.get('SELECT breedId FROM breeds WHERE breedId = ?;', [id]);
+                // returns either id or undefined 
+                return isId? true : false 
+            } catch (err) {
+                console.log(err);
+            }
+        }
+        return getProm();
     }
 
     getBreedById(id) {
-        const breedsCopy = this.getAllBreeds; 
-        const idMatchArray = breedsCopy.filter((breed) => {  // could use findIndexById instead of array.filter
-            return breed['breedId'] == id; 
-        })
-        return idMatchArray[0];
+
+        const dao2 = this.dao;
+        const checkForId = this.checkForId; 
+
+        const getProm = async function () {
+            try {
+                if (!checkForId(id)) {
+                    return null;
+                } else {
+                    const getBreed = await dao2.get('SELECT * FROM breeds WHERE breedId = ?;', [id]);
+                    return getBreed; 
+                };
+            } catch (err) {
+                console.log(err);
+            }
+        }
+        return getProm();
     }
 
     updateBreedById(id, object) {
-        const breedsCopy = this.getAllBreeds; 
-        const index = this.findIndexById(id); // 
-        if (index === -1) {
-            console.log(`index not present: ${index}`);
-            return null;
-        } else {
-            for (let key in object) {
-                if (breedsCopy[index][key] !== object[key]) {
-                    breedsCopy[index][key] = object[key];
-                    console.log(`Key '${key}' successfully updated`);
-                }
+        
+        const dao2 = this.dao;
+        const checkForId = this.checkForId; 
+        const getBreedById = this.getBreedById; 
+
+        const getProm = async function () {
+            try {                
+                if (!checkForId(id)) {
+                    return null;
+                } else {
+                    // iterates through object to create a string of keys and values in format: 'column = value column2 = value2'
+                    let arr = []; 
+                    console.log('object:' + JSON.stringify(object));
+                    for (let key in object) {
+                        arr.push(` ${key} = '${object[key]}'`);   
+                    }
+                    console.log('arr: ' + arr); 
+                    let sql = `UPDATE breeds SET ${arr.join()} WHERE breedId = ?;`
```

Comment:
This is a big no-no as it leaves your application open to [SQL injection](https://en.wikipedia.org/wiki/SQL_injection) - which could result in all your cats being deleted.

It would be much better to explicitly specify each column for the breed (and cat) when updating it. It might seem long-winded in the number of lines of code, but it's well worth being explicit in this case.

Reply:
No, not my cats! But yes, ok.